cmd/credential-providers/ecr/main.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"os"
+
+	ecr "k8s.io/cloud-provider-aws/pkg/providers/v2/credentialprovider/ecr"
```

Comment:
the plugin logic is relatively small and well contained, I would be in favor of keeping all the related code in the main package. 

Also, thoughts on just `cmd/ecr-plugin` or `cmd/ecr-credential-provider` as opposed to having a `credential-providers` sub-directory?

Reply:
I'm all for keeping it in the main honestly. I was reluctant to move them in the first place, so I did it on a separate commit. I can just revert it.

That makes sense to me if we'll only have the ecr plugin for foreseeable future. I created that folder with possibility of other plugins in mind.